pkg/gcloud/client.go

```diff
@@ -2,41 +2,180 @@ package gcloud
 
 import (
 	"context"
+	"fmt"
+	"time"
 
+	"github.com/roppenlabs/silent-assassin/pkg/k8s"
 	"golang.org/x/oauth2/google"
 	compute "google.golang.org/api/compute/v1"
+	container "google.golang.org/api/container/v1"
 )
 
 type GCloudClient struct {
-	Service *compute.Service
+	project                    string
+	location                   string
+	cluster                    string
+	kubeClient                 k8s.IKubernetesClient
+	computeServiceComputeScope *compute.Service
```

Comment:
This makes the functions in gcloud client not testable. We could split the compute part into a separate client called compute client and then expose functions abstracting away the details and inject and mock them here.